pkg/clients/openshift/util.go

```diff
@@ -0,0 +1,95 @@
+package openshift
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"encoding/json"
+	apps "github.com/openshift/api/apps/v1"
+	authorization "github.com/openshift/api/authorization/v1"
+	build "github.com/openshift/api/build/v1"
+	image "github.com/openshift/api/image/v1"
+	route "github.com/openshift/api/route/v1"
+	template "github.com/openshift/api/template/v1"
+)
+
+var (
+	AddToSchemes runtime.SchemeBuilder
+	Scheme       = scheme.Scheme
+	codecs       = serializer.NewCodecFactory(Scheme)
+	decoderFunc  = decoder
+)
+
+func init() {
+	AddToSchemes = append(AddToSchemes,
+		apps.Install,
+		authorization.Install,
+		build.Install,
+		image.Install,
+		route.Install,
+		template.Install,
+	)
+}
+
+func AddToScheme(s *runtime.Scheme) error {
+	return AddToSchemes.AddToScheme(s)
+}
+
+func decoder(gv schema.GroupVersion, codecs serializer.CodecFactory) runtime.Decoder {
+	codec := codecs.UniversalDecoder(gv)
+	return codec
+}
+
+func LoadKubernetesResource(jsonData []byte, namespace string) (runtime.Object, error) {
+	u := unstructured.Unstructured{}
+
+	err := u.UnmarshalJSON(jsonData)
+	if err != nil {
+		return nil, err
+	}
+	u.SetNamespace(namespace)
+	//ToDo Is there a way to register the DeploymentConfig without the Group?
+	if u.GetObjectKind().GroupVersionKind().Kind == "DeploymentConfig" {
+		u.GetObjectKind().SetGroupVersionKind(
+			schema.GroupVersionKind{
+				Version: "v1",
+				Group:   "apps.openshift.io",
+				Kind:    "DeploymentConfig",
+			})
+	}
```

Comment:
This is here because the DC in the fuse yaml doesn't have the full group on the apiVersion (apps.openshift.io/v1) https://github.com/syndesisio/fuse-online-install/blob/1.5.12/resources/fuse-online-operator.yml#L152, and the dynamic controller runtime client we are using requires the full version that has been added to the scheme. 

I had hoped using the legacy sheme option would have got round this, but it didn't work as expected https://github.com/openshift/api/blob/master/apps/v1/legacy.go

Reply:
Is this something we can try add to the Deployment config upstream?

Reply:
Might be worth opening a PR upstream and see if it gets approved. We will still need this here anyway to deal with older/current releases though. I also still think it should be possible to get it registered in the default core group (v1), so might be worth looking into that again at a later stage.